contrib/endpoints/src/api_manager/config_manager.cc

```diff
@@ -0,0 +1,215 @@
+    global_context_->server_config()
+        ->mutable_service_management_config()
+        ->set_refresh_interval_ms(kConfigUpdateCheckInterval);
+  }
+}
+
+void ConfigManager::Init() {
+  if (global_context_->service_name().empty() ||
+      global_context_->config_id().empty()) {
+    GlobalFetchGceMetadata(global_context_, [this](utils::Status status) {
+      on_fetch_metadata(status);
+    });
+  } else {
+    on_fetch_metadata(utils::Status::OK);
+  }
+}
+
+void ConfigManager::on_fetch_metadata(utils::Status status) {
+  if (!status.ok()) {
+    // We should not get here
+    global_context_->env()->LogError("Unexpected status: " + status.ToString());
+    config_roollout_callback_(utils::Status(Code::ABORTED, status.message()),
+                              empty_configs);
+    return;
+  }
+
+  // Update service_name
+  if (global_context_->service_name().empty()) {
+    global_context_->set_service_name(
+        global_context_->gce_metadata()->endpoints_service_name());
+  }
+
+  // Update config_id
+  if (global_context_->config_id().empty()) {
+    global_context_->config_id(
+        global_context_->gce_metadata()->endpoints_service_config_id());
+  }
+
+  // Call ApiManager call with Code::ABORTED, ESP will stop moving forward
+  if (global_context_->service_name().empty()) {
+    std::string msg = "API service name not specified in configuration files";
+    global_context_->env()->LogError(msg);
+    config_roollout_callback_(utils::Status(Code::ABORTED, msg), empty_configs);
+    return;
+  }
+
+  // TODO(jaebong) config_id should not be empty for the first version
+  // This part will be removed after the rollouts feature added
+  if (global_context_->config_id().empty()) {
+    std::string msg = "API config_id not specified in configuration files";
+    global_context_->env()->LogError(msg);
+    config_roollout_callback_(utils::Status(Code::ABORTED, msg), empty_configs);
+    return;
+  }
+
+  // Fetch service account token
+  GlobalFetchServiceAccountToken(global_context_, [this](utils::Status status) {
+    on_fetch_auth_token(status);
+  });
+}
+
+void ConfigManager::on_fetch_auth_token(utils::Status status) {
+  if (!status.ok()) {
+    // We should not get here
+    global_context_->env()->LogError("Unexpected status: " + status.ToString());
+    config_roollout_callback_(utils::Status(Code::ABORTED, status.message()),
+                              empty_configs);
+  }
+
+  if (global_context_->service_account_token()) {
+    // register auth token for servicemanagement services
+    global_context_->service_account_token()->SetAudience(
+        auth::ServiceAccountToken::JWT_TOKEN_FOR_SERVICEMANAGEMENT_SERVICES,
+        service_management_url_ + kServiceManagementServiceManager);
+  }
+
+  // Fetch configs from the Inception
+  auto fetchInfo = std::make_shared<ConfigsFetchInfo>();
+
+  // For now, config manager has only one config_id (100% rollout)
+  fetchInfo->rollouts = {{global_context_->config_id(), 0}};
```

Comment:
I followed the current implementation. Changed it to 100. https://github.com/istio/proxy/blob/master/contrib/endpoints/src/api_manager/api_manager_impl.cc#L62